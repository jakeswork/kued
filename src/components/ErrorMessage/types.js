import { bool } from 'prop-types';

export const propTypes = {
  error: bool,
};

export const defaultProps = {
  error: true,
};
