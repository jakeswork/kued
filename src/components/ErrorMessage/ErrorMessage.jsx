import React, { Fragment } from 'react';
import { FiMessageCircle } from 'react-icons/fi';

import { propTypes, defaultProps } from './types';
import Text from '../Text';
import Button from '../Button';

const ErrorMessage = ({ error }) => error && (
  <Fragment>
    <Text h4 data-test-id="errorHeading">Oops!</Text>
    <Text bold>
      It looks like the server is failing to respond.&nbsp;
      Please wait a moment and try again.
    </Text>
    <br />
    <Text>
      Is this happening regularly?
      Head over to the support
      page and let us know.
    </Text>
    <Button link="/support" secondary icon={<FiMessageCircle />}>
      Support
    </Button>
  </Fragment>
);

ErrorMessage.propTypes = propTypes;

ErrorMessage.defaultProps = defaultProps;

export default ErrorMessage;
