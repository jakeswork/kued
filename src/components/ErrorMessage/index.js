import withStyles from 'react-jss';

import ErrorMessage from './ErrorMessage';
import styles from './styles';

export default withStyles(styles)(ErrorMessage);
