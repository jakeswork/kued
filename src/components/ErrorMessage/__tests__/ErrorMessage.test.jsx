import React from 'react';
import { shallow } from 'enzyme';

import ErrorMessage from '../ErrorMessage';

const defaultProps = {};

describe('The ErrorMessage instance', () => {
  describe('When rendered with default props', () => {
    it('should render without throwing an error', () => {
      const wrapper = shallow(<ErrorMessage {...defaultProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('When the error prop is false', () => {
    it('Should render null', () => {
      const newProps = {
        ...defaultProps,
        error: false,
      };
      const wrapper = shallow(<ErrorMessage {...newProps} />);

      expect(wrapper).toMatchSnapshot();
    });
  });
});
