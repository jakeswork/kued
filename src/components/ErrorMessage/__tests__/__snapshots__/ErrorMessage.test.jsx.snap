// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ErrorMessage instance When rendered with default props should render without throwing an error 1`] = `
<Fragment>
  <Jss(Text)
    bold={false}
    caption={false}
    data-test-id="errorHeading"
    h1={false}
    h2={false}
    h3={false}
    h4={true}
    h5={false}
    link={null}
    style={Object {}}
  >
    Oops!
  </Jss(Text)>
  <Jss(Text)
    bold={true}
    caption={false}
    h1={false}
    h2={false}
    h3={false}
    h4={false}
    h5={false}
    link={null}
    style={Object {}}
  >
    It looks like the server is failing to respond.Â  Please wait a moment and try again.
  </Jss(Text)>
  <br />
  <Jss(Text)
    bold={false}
    caption={false}
    h1={false}
    h2={false}
    h3={false}
    h4={false}
    h5={false}
    link={null}
    style={Object {}}
  >
    Is this happening regularly? Head over to the support page and let us know.
  </Jss(Text)>
  <Jss(Button)
    analyticsLabel={null}
    danger={false}
    flat={false}
    icon={<FiMessageCircle />}
    link="/support"
    secondary={true}
    success={false}
  >
    Support
  </Jss(Button)>
</Fragment>
`;
